File: searching algoritms/linear_search.rb
#begin Write a function called linearSearch which accepts an array and a value, and returns the index at which the value exists. If the value does not exist in the array, return -1.

#Don't use indexOf to implement this function!

def linear_search arr, value
  arr.each_with_index do |num, index|
    if num == value
      return index
    end
  end

  return -1
end

p linear_search([1,2,3], 5)

File: searching algoritms/linear-search.js
/* Write a function called linearSearch which accepts an array and a value, and returns the index at which the value exists. If the value does not exist in the array, return -1.

Don't use indexOf to implement this function! */

const linearSearch = (arr, value) => {
  
  for(let i = 0; i < arr.length; i++) {
    if(arr[i] == value){
      return i
    }
  }

  return -1
}

console.log(linearSearch([10, 15, 20, 25, 30], 15)) // 1
console.log(linearSearch([9, 8, 7, 6, 5, 4, 3, 2, 1, 0], 4)) // 5
console.log(linearSearch([100], 100)) // 0
console.log(linearSearch([1,2,3,4,5], 6)) // -1
console.log(linearSearch([9, 8, 7, 6, 5, 4, 3, 2, 1, 0], 10)) // -1
console.log(linearSearch([100], 200)) // -1